src/features/controls.ts

```diff
@@ -498,6 +499,20 @@ export const ControlsMixin = <T extends Constructor<ModelViewerElementBase>>(
       controls.updateAspect(this[$scene].aspect);
       this.requestUpdate('fieldOfView', this.fieldOfView);
       controls.jumpToGoal();
+
+      // Immediately queue a render to happen at microtask timing. This is
+      // necessary because setting the width and height of the canvas has the
+      // side-effect of clearing it, and also if we wait for the next rAF to
+      // render again we might get hit with yet-another-resize, or worse we
+      // may not actually be marked as dirty and so render will just not
+      // happen. Queuing a render to happen here means we will render twice on
+      // a resize frame, but it avoids most of the visual artifacts associated
+      // with other potential mitigations for this problem. See discussion in
+      // https://github.com/GoogleWebComponents/model-viewer/pull/619 for
+      // additional considerations.
+      Promise.resolve().then(() => {
```

Comment:
It looks like moving this doesn't have a practical difference other than changing where the implementation lives. Both methods (`ModelViewer.prototype[$onResize]` and `ModelScene.prototype.setSize`) are invoked in the same turn of the event loop, and a microtask will always happen after the callstack for that turn unwinds.

`ModelScene` is always a member of a `ModelViewerElementBase`-derived class, `ControlsMixin` is optional implementation. So, with this change, any class that does not use `ControlsMixin` will inherit the render issue described in the comment above.

Reply:
Interesting; this was how I fixed the resize jank. It seemed to be drawing a new frame before it figured out the the zoom level. It still happens a little though, even after this change, so there must be a better way. I need to find a better way to debug the event loop.

Reply:
It's possible that there are cases where one method is called and the other isn't. I'll dig deeper and see if that's what is going on.